markdown/docs/gomemlimit.mdx

```diff
@@ -0,0 +1,205 @@
+- **Tracks Heap Memory, Not Total Memory** - GOMEMLIMIT tracks heap memory only. The total process memory usage also includes stack memory, code memory, and more. Be cautious when interpreting memory measurements to avoid inaccurate conclusions.
+
+### Diving Deeper
+
+For those who want to explore the implementation details, check out the Go runtime source code: [runtime/mgcpacer.go](https://github.com/golang/go/blob/3bd08b97921826c1b0a5fbf0789f4b49d7619977/src/runtime/mgcpacer.go#L966C29-L966C48).
+
+### Choosing the right value for GOMEMLIMIT
+
+Deciding to use GOMEMLIMIT is straightforward. Choosing the right value? Not so much. 
+The optimal value is inherently subjective, dependent on your application’s behavior, traffic patterns, and performance goals.
+
+To guide you through this process, here are the key considerations:
+
+- **Memory Guarantees** - How much memory can you confidently allocate to the application without risking an OOM (Out of Memory) event? In kubernetes, this value is controlled by the memory resource request in the pod manifest.
+- **Worst-Case Memory Usage** - What is the maximum memory your application might consume? This is often unbounded in real-world scenarios, and is hard to calculate and predict accurately.
+- **Crash Tolerance** - How tolerant is your application to crashes? For example, under extreme load, do you prioritize stability with larger safety margins, or do you prefer to maximize resource efficiency?
+
+These factors are not formulaic! the Go runtime’s GC logic is too complex to optimize with precision. Instead, selecting a GOMEMLIMIT value involves a series of trade-offs between performance, memory efficiency, and stability.
+However, there are few observations that can help you in choosing a value:
+
+Key Trade-Offs:
+
+- **Memory Reservation** - ensure your platform (e.g., Kubernetes) reserves a guaranteed amount of memory for your application. Without this, an OOM Kill event could occur unpredictably, undermining stability regardless of the GOMEMLIMIT value.
+- **Buffer for Stability** - GOMEMLIMIT should be less than the total memory allocated to your application. This accounts for:
+    - Memory not tracked by GOMEMLIMIT (e.g., stack memory, code memory).
+    - A buffer to accommodate GC delays when memory usage crosses the limit.
+    - safety margin.
+- **Low vs High GOMEMLIMIT**:
+    - **Lower Value** - Provides greater stability by allocating more memory headroom, reducing the risk of crashes.
+    - **Higher Value** - Improves memory efficiency by prioritizing business logic over safety margins, but increases the risk of hitting OOM under load.
+
+In our setup, we followed the [OpenTelemetry memory limiter processor best practices](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/memorylimiterprocessor#best-practices):
```

Comment:
It is the first time Opentelemetry is mention, in general seems like we hook our use case without context. maybe worth adding another section of our case after the general explanations about GOMEMLIMIT?